gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/GobblinOutputFormat.java

```diff
@@ -0,0 +1,151 @@
+package gobblin.runtime.mapreduce;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.source.workunit.WorkUnit;
+import gobblin.util.ForkOperatorUtils;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.JobStatus;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.output.NullOutputFormat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Closer;
+
+/**
+ * Hadoop {@link org.apache.hadoop.mapreduce.OutputFormat} implementation with a custom {@link OutputCommitter}
+ */
+public class GobblinOutputFormat extends NullOutputFormat<NullWritable, NullWritable> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GobblinOutputFormat.class);
+
+  @Override
+  public OutputCommitter getOutputCommitter(TaskAttemptContext context) {
+    return new GobblinOutputCommitter();
+  }
+
+  /**
+   * Hadoop {@link OutputCommitter} implementation that overrides the default
+   * {@link #abortJob(JobContext, org.apache.hadoop.mapreduce.JobStatus.State)} behavior. This is necessary to add
+   * functionality for cleaning up staging data when the
+   * {@link com.linkedin.uif.runtime.JobLauncher#cancelJob(Properties)} method is called via Azkaban. Azkaban only
+   * allows the cancel method run to for 5 ms until it does a hard kill on the process, so instead, we will clean up the
+   * staging data in the AM.
+   */
+  private class GobblinOutputCommitter extends OutputCommitter {
+
+    @Override
+    public void abortJob(JobContext jobContext, JobStatus.State state) throws IOException {
+      LOG.info("Aborting Job: " + jobContext.getJobID());
+
+      Configuration conf = jobContext.getConfiguration();
+
+      URI fsUri = URI.create(conf.get(ConfigurationKeys.FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI));
+      FileSystem fs = FileSystem.get(fsUri, conf);
+
+      Path mrJobDir =
+          new Path(conf.get(ConfigurationKeys.MR_JOB_ROOT_DIR_KEY), conf.get(ConfigurationKeys.JOB_NAME_KEY));
+      Path jobInputDir = new Path(mrJobDir, "input");
+
+      for (FileStatus status : fs.listStatus(jobInputDir)) {
+
+        Closer workUnitFileCloser = Closer.create();
+        if (status.getPath().getName().endsWith(".wu")) {
+
+          WorkUnit wu = new WorkUnit();
+          wu.readFields(workUnitFileCloser.register(new DataInputStream(fs.open(status.getPath()))));
```

Comment:
I can't for a DataInputStream, also its just a serialized WorkUnit object, not a text file.